src/neo/SmartContract/Native/LedgerContract.cs

```diff
@@ -220,7 +225,9 @@ public Header GetHeader(DataCache snapshot, uint index)
         /// <returns>The <see cref="TransactionState"/> with the specified hash.</returns>
         public TransactionState GetTransactionState(DataCache snapshot, UInt256 hash)
         {
-            return snapshot.TryGet(CreateStorageKey(Prefix_Transaction).Add(hash))?.GetInteroperable<TransactionState>();
+            var state = snapshot.TryGet(CreateStorageKey(Prefix_Transaction).Add(hash))?.GetInteroperable<TransactionState>();
+            if (state is not null && state.Trimmed) return null;
+            return state;
```

Comment:
Here we return only non-trimmed transactions to the stack. In fact, Trimmed is a part of internal implementation, the outside user shouldn't know about this field. I suggest not to change user-faced wrapper of transaction.